pydicom/dataset.py

```diff
@@ -1089,14 +1089,30 @@ def clear(self):
         """Delete all data elements."""
         self._dict.clear()
 
-    def pop(self, *args, **kwargs):
-        return self._dict.pop(*args, **kwargs)
+    @staticmethod
+    def _handle_tuple_and_keyword(fct, key, *args):
+        if isinstance(key, (str, compat.text_type)):
+            tag = tag_for_keyword(key)
+        else:
+            try:
+                tag = Tag(key)
+            except (ValueError, OverflowError):
+                return fct(key, *args)
+        return fct(tag, *args)
+
+    def pop(self, key, *args):
+        """Emulate dictionary `pop`, but additionally support tag ID tuple
+        and DICOM keyword."""
+        return self._handle_tuple_and_keyword(self._dict.pop, key, *args)
 
     def popitem(self):
         return self._dict.popitem()
 
-    def setdefault(self,  *args, **kwargs):
-        return self._dict.setdefault(*args, **kwargs)
+    def setdefault(self, key, *args):
+        """Emulate dictionary `setdefault`, but additionally support tag ID
+        tuple and DICOM keyword."""
+        return self._handle_tuple_and_keyword(
+            self._dict.setdefault, key, *args)
 
```

Comment:
I'm rethinking self._dict.setdefault as the fallback function here.  It was already in the code that is being extended in this PR, but it gives the user access to the 'hidden' self._dict without checks.  Perhaps the code should at least check that the value passed is a DataElement instance?

Also, for documentation for the function, I'm wondering if we could give an example.  The only one I could think of that is useful is to setdefault a Sequence before assigning something to it (untested):  `ds.setdefault("PatientSetupSequence", Sequence()).value.PatientPosition = "HFS"`

Reply:
Realized my example wouldn't work as Seuqence() is not a DataElement...  but another example is to set a default patient name before printing info, for example.  `ds.setdefault("PatientName", "")`

Reply:
> it gives the user access to the 'hidden' self._dict without checks

I agree that this is not good - haven't thought about this. I now changed `setdefault` to also accept values (e.g. all default values that are not a `Dataset` are considered a `Dataset` value). There is still the possibility to set a default value not suitable for the given tag, but you can also do this by creating a `DataElement` manually. 
Please check if this makes sense...